interview-questions.md

```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
 Your answer:
+A hash in Ruby is like an array in JavaScript because it stores data in key-value pairs, but hashes don't use numbers as an index.
 
 Researched answer:
+Hashes can also be look at as dictionaries because they perform like them. The term iterate and its definition are an example of a hash:
+iterate - To execute a set of actions repeatedly to accomplish a task.
+In this example, the term "iterate" is the key and the definition that follows is the value. We access the data by looking up the word in the dictionary
 
 2. What is a gem?
 
 Your answer:
+A gem is Ruby code that has been made available for developers to use. Ruby on Rails is an example of a Ruby gem. Rails is a package of Ruby code that is made available to use in the development of an application. 
 
 Researched answer:
+Throughout our challenges over this past week of the cohort, we were using Ruby on Rails packages as we were creating models, working with schemas, and running migrations in our database manipulations. This particular gem provides the essential built-in code needed to connect to and work with backend portion of the application.
```

Comment:
more content can be to state a ruby gem that you use and talk about what it added to the application